efi/fwupdate.c

```diff
@@ -454,10 +454,6 @@ fwup_apply_capsules(EFI_CAPSULE_HEADER **capsules,
 
 	rc = uefi_call_wrapper(RT->QueryCapsuleCapabilities, 4, capsules,
 				num_updates, &max_capsule_size, reset);
-	if (EFI_ERROR(rc)) {
-		fwup_warning(L"Could not query capsule capabilities: %r", rc);
-		return rc;
-	}
 	fwup_debug(L"QueryCapsuleCapabilities: %r max: %ld reset:%d",
```

Comment:
you shouldn't be outputting these statements if the previous call failed.  If this really is something we're going to support, it needs some other changes:
1. Only output this debug string when EFI_SUCCESS
2. Need to ensure some other way that `EfiResetWarm` works all the time

Reply:
Yup, I've asked HP for clarification; it might be we're passing things to QueryCapsuleCapabilities() in the "wrong" way.